apps/api/views/chat.py

```diff
@@ -0,0 +1,339 @@
+
+
+def check_experiment_access(request, experiment, participant_id):
+    """
+    Check if the request has access to the experiment based on public API settings.
+
+    Returns:
+        Response object if access denied, None if access allowed
+    """
+    if request.team and experiment.team != request.team:
+        # Authenticated requests must be constrained to their team
+        return Response({"error": "Access denied"}, status=status.HTTP_403_FORBIDDEN)
+
+    if not experiment.is_public and not experiment.is_participant_allowed(participant_id):
+        return Response({"error": "Access denied"}, status=status.HTTP_403_FORBIDDEN)
+
+    return None
+
+
+def check_session_access(request, session):
+    """
+    Check if the request has access to the session based on experiment's public API settings.
+
+    Returns:
+        Response object if access denied, None if access allowed
+    """
+    return check_experiment_access(request, session.experiment, session.participant.identifier)
+
+
+@extend_schema(
+    operation_id="chat_start_session",
+    summary="Start a new chat session for a widget",
+    tags=["Chat"],
+    request=ChatStartSessionRequest,
+    responses={201: ChatStartSessionResponse},
+    # auth=["{}"],
+    examples=[
+        OpenApiExample(
+            name="StartChatSession",
+            summary="Start a new chat session for an experiment",
+            value={
+                "experiment_id": "123e4567-e89b-12d3-a456-426614174000",
+                "session_data": {"source": "widget", "page_url": "https://example.com"},
+            },
+        ),
+    ],
+)
+@api_view(["POST"])
+@authentication_classes(AUTH_CLASSES)
+@permission_classes([])
+def chat_start_session(request):
+    """
+    Start a new chat session for a widget
+    """
+    serializer = ChatStartSessionRequest(data=request.data)
+    serializer.is_valid(raise_exception=True)
+
+    data = serializer.validated_data
+    experiment_id = data["chatbot_id"]
+    participant_id = data.get("participant_id")
+    session_data = data.get("session_data", {})
+
+    # First, check if this is a public experiment
+    experiment = get_object_or_404(Experiment, public_id=experiment_id)
+    if not experiment.is_working_version:
+        return Response(
+            {"error": "Chatbot ID must reference the working version of an chatbot"},
+            status=status.HTTP_400_BAD_REQUEST,
+        )
+
+    access_response = check_experiment_access(request, experiment, participant_id)
+    if access_response:
+        return access_response
+
+    if request.user.is_authenticated:
+        team = request.team
+        user = request.user
+    else:
+        team = experiment.team
+        user = None
+
+    # Create or get participant
+    if user is not None:
+        participant_id = participant_id or user.email
+        if participant_id != user.email:
+            # TODO: re-evaluate this, it doesn't seem correct in this instance
```

Comment:
wouldn't this cover the cases we are trying to catch?

> if user is not None:
>     if participant_id and participant_id != user.email:
>         return Response({"error": "Participant ID must match your email...)
>     participant_id = participant_id or user.email

Reply:
The current logic works but it doesn't feel quite correct for an API. It matches what we do for web sessions but if another systems uses this to integrate chat into their app then they should be able to specify the participant ID (I think)